### DISCUSSION TITLE
Help Needed: Updating Image from 22.4.35 to 22.4.46 with Data Persistence

### DISCUSSION THREAD

**Discussion by devMarc5598** [immauss/openvas#277] [General]
Hello,
I'm currently trying to update my image from version 22.4.35 to 22.4.46 and wish to retain my current data during the update. However, I'm encountering permission errors while attempting to build a container using my existing data with the new image. 

I need some advise on how to resolve these permission issues, or provide any best practices for upgrading the image while maintaining data integrity?

Some of my logs while starting the container the first time:
> Checking for existing DB
> Running first start configuration...
> NEWDB=false
> LOADDEFAULT=false
> Checking DB Version
> Current GVMd database version is  255
> Migrate the database if needed.
> Updating NVTs and other data
> This could take a while if you are not using persistent storage for your NVTs
>  or this is the first time pulling to your persistent storage.
>  the time will be mostly dependent on your available bandwidth.
> Checking age of current data feeds from Greenbone.
> ImageFeeds=1715999411
> InstalledFeeds=1700649551
> Updating local feeds with newer image feeds.
> Syncing all feeds from GB
> Synchronizing the Notus feed from Immauss Cybersecurity
> And all others from the GB Community feed
> Running as root. Switching to user 'gvm' and group 'gvm'.
> ❌Error: Permission error while trying to open the lock file 
> /var/lib/openvas/feed-update.lock

Thank you in advance for your help!

Best regards,
Marc

**Comment by immauss** [comment]
Marc,

    This "shouldn't" happen. 

     Can you give me a little more information? 

     Is this on a docker volume or local mount ?
     Did you start from command line, or use docker-compose? ( please provide the command line or the compose file)

    With that info, I should be able to help you work it out. 

-Scott

**Reply by devMarc5598** [reply]
Hello,
Thank you for your quick response.

I wanted to let you know that I have resolved the issue. The problem was due to a line in my Dockerfile:


```
FROM immauss/openvas:22.4.46-slim
RUN groupadd -r gvm && useradd -r -g gvm gvm
```

I had added this user line in an older image to create the user 'gvm' for some automation scripts. However, this caused the user to have a different ID than 1000, resulting in permission issues. I have now removed this line, and the user 'gvm' now has the correct permissions.

Best regards,
Marc
